torchgeo/datasets/utils.py

```diff
@@ -198,7 +198,7 @@ def download_radiant_mlhub_collection(
     collection.download(output_dir=download_root, api_key=api_key)
 
 
-@dataclass(frozen=True)
+@dataclass
```

Comment:
pytorch-lightning seems to want to overwrite the values in all objects in a sample, including the bounding box. I'm not sure why it needs to do this, but if I don't remove `frozen=True`, I see the following testing error:
```
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py:737: in fit
    self._call_and_handle_interrupt(
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py:682: in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py:772: in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py:1195: in _run
    self._dispatch()
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py:1274: in _dispatch
    self.training_type_plugin.start_training(self)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/plugins/training_type/training_type_plugin.py:202: in start_training
    self._results = trainer.run_stage()
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py:1284: in run_stage
    return self._run_train()
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py:1314: in _run_train
    self.fit_loop.run()
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/loops/base.py:145: in run
    self.advance(*args, **kwargs)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/loops/fit_loop.py:234: in advance
    self.epoch_loop.run(data_fetcher)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/loops/base.py:145: in run
    self.advance(*args, **kwargs)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/loops/epoch/training_epoch_loop.py:160: in advance
    batch = self.trainer.accelerator.batch_to_device(batch)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/accelerators/accelerator.py:206: in batch_to_device
    return model._apply_batch_transfer_handler(batch, device=device, dataloader_idx=dataloader_idx)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/core/lightning.py:291: in _apply_batch_transfer_handler
    batch = self.transfer_batch_to_device(batch, device, dataloader_idx)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/core/hooks.py:707: in transfer_batch_to_device
    return move_data_to_device(batch, device)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/utilities/apply_func.py:279: in move_data_to_device
    return apply_to_collection(batch, dtype=dtype, function=batch_to)
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/utilities/apply_func.py:100: in apply_to_collection
    v = apply_to_collection(
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/utilities/apply_func.py:114: in apply_to_collection
    v = apply_to_collection(
../.spack/.spack-env/view/lib/python3.8/site-packages/pytorch_lightning/utilities/apply_func.py:146: in apply_to_collection
    setattr(result, field_name, v)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = BoundingBox(minx=-8411085.25243577, maxx=-8410957.25243577, miny=4720865.81310193, maxy=4720993.81310193, mint=0.0, maxt=9.223372036854776e+18), name = 'minx'
value = -8411085.25243577

>   ???
E   dataclasses.FrozenInstanceError: cannot assign to field 'minx'
```